apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-preview-dropdown-selected.js

```diff
@@ -0,0 +1,30 @@
+import { __ } from '@wordpress/i18n';
+import tracksRecordEvent from './track-record-event';
+
+/**
+ * Return the event definition object to track `wpcom_block_editor_template_part_detach_blocks`.
+ * @returns {import('./types').DelegateEventHandler} event object definition.
+ */
+export default () => ( {
+	id: 'wpcom-preview-dropdown-selected',
+	selector: '.components-menu-item__button',
+	type: 'click',
+	capture: true,
+	handler: ( _event, target ) => {
+		const mapTextToPreviewMode = {
+			[ __( 'Desktop' ) ]: 'desktop',
+			[ __( 'Desktop (50%)' ) ]: 'zoom-out',
+			[ __( 'Tablet' ) ]: 'tablet',
+			[ __( 'Mobile' ) ]: 'mobile',
+			[ `${ __( 'View site' ) }\n${ __( '(opens in a new tab)' ) }` ]: 'view-site',
+		};
+
+		const item = target.querySelector( '.components-menu-item__item' );
+		const previewMode = mapTextToPreviewMode[ item?.innerText ];
+		if ( previewMode ) {
+			tracksRecordEvent( 'wpcom_preview_dropdown_selected', {
+				preview_mode: previewMode,
+			} );
```

Comment:
This can run for both the post editor and the site editor. Can we include an additional property to flag which editor is being used?

Also, a nitpick: because these are previews from the editor, I'd rename the event name to `wpcom_editor_preview_dropdown_selected`. Just in case we have in the future a similar dropdown to preview stuff outside of the editor.

Reply:
> This can run for both the post editor and the site editor. Can we include an additional property to flag which editor is being used?

it's tracked by the `editor_type`.

> I'd rename the event name to wpcom_editor_preview_dropdown_selected. Just in case we have in the future a similar dropdown to preview stuff outside of the editor.

Okay!

Reply:
> it's tracked by the editor_type.

Oh! Didn't realize the `tracksRecordEvent` function was already doing that. Sorry!